src/tools/pub/cmd/pub-outdated.md

```diff
@@ -0,0 +1,236 @@
+---
+title: pub outdated
+description: Use pub outdated to help you update your package dependencies.
+diff2html: true
+---
+
+_Outdated_ is one of the commands of the [pub tool](/tools/pub/cmd).
+
+{% prettify nocode tag=pre+code %}
+$ pub outdated [options]
+{% endprettify %}
+
+Use `pub outdated` to identify out-of-date [package dependencies][]
+and get advice on how to update them.
+[Best practices for dependency management][best practices]
+include using the most recent stable package versions,
+so you can get the latest bug fixes and improvements.
+
+{{site.alert.version-note}}
+  The `pub outdated` command was introduced in Dart 2.8.
+{{site.alert.end}}
+
+## Overview
+
+Here's how you can use `pub outdated` to help you
+update the dependencies of a package that you own
+(whether it's an app or library package):
+
+1. If your package doesn't have a `pubspec.lock` file
+   checked into source control,
+   **run `pub get`** in the top directory of the package —
+   the directory that contains your package's
+  [`pubspec.yaml`](/tools/pub/pubspec) file.
+1. **Run `pub outdated`**
+   to identify which package dependencies are out-of-date.
+   Note the affected packages,
+   so that later you can test the behavior of code that uses them.
+1. Follow the recommendations of `pub outdated` for updating the packages.
+   Some updates might require only running `pub upgrade`.
+   Others might require updating `pubspec.yaml` before running `pub upgrade`.
+1. **Run `pub outdated`** to confirm that you've updated all the dependencies.
+1. **Test** your package to confirm that it still works as expected.
+
+
+## Example
+
+Here's an example of running `pub outdated` on
+[PENDING: link to source code]:
+
+```terminal
+$ pub outdated
+Package              Current   Upgradable  Resolvable  Latest    
+dependencies: all up-to-date
+
+dev_dependencies    
+build_web_compilers  1.2.3     1.2.3       2.10.0      2.10.0    
+
+transitive dependencies
+_fe_analyzer_shared  -         -           2.0.0       2.0.0     
+analyzer             0.35.4    0.35.4      0.39.5      0.39.5    
+...
+
+9  dependencies are constrained to versions that are older than a resolvable version.
+To update these dependencies, edit pubspec.yaml.
+```
+
+<span style="color:red">[PENDING: Use a different example.
+Link to it so it's reproduceable.
+Update the text to match.]</span>
+
+The preceding output shows that all regular dependencies are up-to-date,
+but one [dev dependency][] (`build_web_compilers`) is out-of-date.
+Several [transitive dependencies][] are also out-of-date.
+For details on what each column shows, see the
+[output columns](#output-columns) section of this page.
+
+To fix the dev dependency,
```

Comment:
I think it might be good if the example shows the three main cases:

  1. A dependency is outdated, but can be moved to latest by running `pub upgrade`.
  1. A dependency is outdated, but pubspec.yaml doesn't allow the latest version, so an edit to pubspec is needed
  1. A dependency is outdated, but the app is on the latest one that's mutually compatible

I personally find it easiest to explain this via the columns (hence why I lead with those in the blog post), which correspond to the three cases above like this:

First, the colors in the output are important here. They show the user where to start (the left-most non-red column)

  1. Current version is red, upgradeable is bold white, upgradeable/resolveable/latest show the same number: outdated but can be upgraded automatically; the dependency needs a `pub upgrade`
  1. Upgradeable is red, resolvable/latest show the same number: outdated and not automatically upgradeable; edit pubspec.yaml
  1. Resolvable is red: The app is already locked on the latest version that's mutually compatible. Nothing the developer can do (expect to file bugs on the packages that cause the mutual exclusion)

Reply:
OK, sounds like I should feature a screenshot so the colors are visible.

Reply:
I... didn't know how to do the 3rd one. You can see the code I used at https://github.com/kwalrath/po-example.

Reply:
See https://github.com/kwalrath/po-example/pull/1